src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -381,6 +383,135 @@ public Task GetMigratedServerUrls()
+
+                if (migrationConfiguration.IsExportCommand == false)
+                {
+                    try
+                    {
+                        var line = process.StandardOutput.ReadLine();
+                        // return type is json, will throw if we fail to deserialize
+                        JsonConvert.DeserializeObject(line);
+                        using (var sw = new StreamWriter(ResponseBodyStream()))
+                        {
+                            await sw.WriteAsync(line);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        var error = process.StandardError.ReadLine();
+                        process.Kill();
+                        throw new InvalidOperationException($"Process error: {error}, exception: {e}");
+                    }
+
+                    return;
+                }
+
+                var operationId = GetLongQueryString("operationId", false) ?? Database.Operations.GetNextOperationId();
+                var token = CreateOperationToken();
+                var transformScript = migrationConfiguration.TransformScript;
+
+                var t = Database.Operations.AddOperation(Database, $"Migration from: {migrationConfiguration.DatabaseTypeName}",
+                    Operations.OperationType.DatabaseMigration,
+                    onProgress =>
+                    {
+                        return Task.Run(() =>
+                        {
+                            var result = new SmugglerResult();
+
+                            try
+                            {
+                                using (ContextPool.AllocateOperationContext(out DocumentsOperationContext migrateContext))
+                                using (var stream = new GZipStream(process.StandardOutput.BaseStream, CompressionMode.Decompress))
+                                {
+                                    var options = new DatabaseSmugglerOptionsServerSide
+                                    {
+                                        TransformScript = transformScript
+                                    };
+                                    DoImportInternal(migrateContext, stream, options, result, onProgress, token);
+                                }
+                            }
+                            catch (OperationCanceledException)
+                            {
+                                process.Kill();
+                                throw;
+                            }
+                            catch (ObjectDisposedException)
+                            {
+                                process.Kill();
+                                throw;
+                            }
+                            catch (Exception e)
+                            {
+                                var error = process.StandardError.ReadLine();
```

Comment:
You need to read whole error here. Check how this is done in Embedded for example.